### ISSUE TITLE
PAPI counters don't work

### ISSUE THREAD

**Issue by msimberg** [STEllAR-GROUP/hpx#4820]
Unsure if this is a system, PAPI, APEX, or HPX issue, so looking for input from e.g. @khuck and @biddisco.

Currently our PAPI tests fail like this on Piz Daint: https://cdash.cscs.ch/testDetails.php?test=23719608&build=115247. As far as I know nothing has changed in our PAPI counters for a long while (except build-related options). That particular example asks for `PAPI_FP_INS` which `--hpx:papi-event-info=all` does not list, `papi_avail` also lists it as not available, so in that sense the failure is reasonable. However, I've tried with other counters that are listed as available and the same error appears. The exception is thrown by the PAPI performance counters in HPX but it seems like APEX is unhappy with something before that as well.

Has anyone tried the PAPI counters recently? This does smell like it's just a problem with PAPI on Piz Daint.

**Comment by khuck** [comment]
Hmmm...there were some recent changes to `proc_read.cpp` but they are completely unrelated to this error.  Can you point me to some instructions on how I can reproduce this configuration?

**Comment by msimberg** [comment]
> Hmmm...there were some recent changes to `proc_read.cpp` but I don't think they are related to this error. Can you point me to some instructions on how I can reproduce this configuration? The error from APEX is trying to enable RAPL power metrics that Piz Daint may have recently enabled in their PAPI configuration. If you try `papi_native_avail` you'll see if the RAPL component has been enabled.

The CMake configuration options are here: https://cdash.cscs.ch/buildSummary.php?buildid=115247#!#Configure. I don't think the compiler is relevant, it's clang 8. The PAPI version may be relevant (5.7.0.2). Otherwise it's a normal build with C++17, `HPX_WITH_APEX=ON`, `HPX_WITH_PAPI=ON`, `APEX_WITH_PAPI=ON`. If you're able to use the PAPI counters normally in your build I would say this is just a configuration issue.

As for RAPL, `papi_native_avail` says this:
```
 Native Events in Component: cray_rapl                                                                                          
===============================================================================                                                 
| cray_rapl:::UNITS:POWER                                                      |                                                
|            Scaling factor for power (internal)                               |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::UNITS:ENERGY                                                     |                                                
|            Scaling factor for energy (internal)                              |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::UNITS:TIME                                                       |                                                
|            Scaling factor for time (internal)                                |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::THERMAL_SPEC                                                     |                                                
|            Thermal specification                                             |                                                
|     Units: W                                                                 |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::MINIMUM_POWER                                                    |                                                
|            Minimum power                                                     |                                                
|     Units: W                                                                 |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::MAXIMUM_POWER                                                    |                                                
|            Maximum power                                                     |                                                
|     Units: W                                                                 |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::MAXIMUM_TIME_WINDOW                                              |                                                
|            Maximum time window                                               |                                                
|     Units: s                                                                 |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::PACKAGE_ENERGY                                                   |                                                
|            Energy used by chip package                                       |                                                
|     Units: nJ                                                                |                                                
--------------------------------------------------------------------------------                                                
| cray_rapl:::DRAM_ENERGY                                                      |                                                
|            Energy used by DRAM in package                                    |                                                
|     Units: nJ                                                                |                                                
===============================================================================                                                 
```
which I guess looks ok. Based on the error messages I don't think RAPL is actually the problem here.

**Comment by khuck** [comment]
I think I agree that APEX isn't causing the exception.  When APEX encounters that problem getting a RAPL counter name, it just continues on to the next one: https://github.com/khuck/xpress-apex/blob/c59fac98a25230b8909a7bbfd062f37ae4099dd7/src/apex/proc_read.cpp#L122-L131 (I just updated that code to give more error information).  You could try toggling `HPX_WITH_PAPI=ON, APEX_WITH_PAPI=OFF` and `HPX_WITH_PAPI=OFF, APEX_WITH_PAPI=ON` to see which library is triggering the exception.  It is possible that both libraries are trying to use an exclusive resource within PAPI.

**Comment by msimberg** [comment]
I can confirm that APEX is unrelated to this, and things still work in 1.4.1, so something has gone wrong on master along the way. I'll try to fix this before 1.5.0 (not necessarily for the first release candidate).
